src/schema/sources.ts

```diff
@@ -1543,7 +1554,26 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       info,
     ): Promise<GQLSource> => {
       await ensureSourcePermissions(ctx, id);
-      return getSourceById(ctx, info, id);
+      const source = await getSourceById(ctx, info, id);
+
+      if (!source.moderationRequired || !source.currentMember) {
+        return source;
+      }
+
+      const isModerator =
+        sourceRoleRank[source.currentMember.role] >= sourceRoleRank.moderator;
+      const status = SourcePostModerationStatus.Pending;
+      const query: FindOptionsWhere<SourcePostModeration> = {
+        status,
+        sourceId: source.id,
+        ...(!isModerator && { createdById: ctx.userId }),
+      };
+
+      const moderationPostCount = await ctx.con
+        .getRepository(SourcePostModeration)
+        .countBy(query);
+
+      return { ...source, moderationPostCount };
```

Comment:
Realized just now what you were doing. 

We usually add the fields through graphorm, the difference is that if you request GQLSource anywhere graphorm will resolve the field, like this it only works on this specific query.

But since this is count I am fine with adding it here for now. Probably ideal to add this to flags as well so its not counted each time but let's not complicate for now since it was defined like this in RFC.